src/process/status.cr

```diff
@@ -5,12 +5,38 @@ class Process::Status
   def initialize(@exit_status)
   end
 
+  # Returns `true` if the process was terminated by a signal.
+  def signal_exit?
+    #define __WIFSIGNALED(status) (((signed char) (((status) & 0x7f) + 1) >> 1) > 0)
+    ((LibC::Char.new(@exit_status & 0x7f) + 1) >> 1) > 0
```

Comment:
`LibC::Char` is unsigned. `LibC::UChar` should be used instead.

In Mac, this macro is defined like this (actual definition depends on other macros):
`status != 0x7f && status != 0`

I think is the same, but I'm having trouble with the math to confirm :-)